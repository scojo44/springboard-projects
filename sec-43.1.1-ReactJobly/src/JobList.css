.JobList {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: 0 auto;
}
.JobList ul {
  flex: 1;
  margin: 0;
  padding: 0;
  overflow-y: scroll;
}
