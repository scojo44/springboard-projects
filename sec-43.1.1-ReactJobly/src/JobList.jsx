import React from 'react'
import JoblyApi from './api'
import useJoblyAPI from './hooks/useJoblyAPI'
import LoadingIndicator from './LoadingIndicator'
import Alert from './Alert'
import SearchForm from './SearchForm'
import JobCard from './JobCard'
import './JobList.css'

export default function JobList() {
  const {data: jobs, setParams, error, isLoading} = useJoblyAPI(JoblyApi.getJobs);

  if(isLoading) return <LoadingIndicator />;

  return (
    <section className="JobList">
      <SearchForm fieldName="titleLike" returnQuery={query => setParams(query)} />
      {error
        ? <Alert type="error" messages={[`Error loading jobs: ${error}`]} />
        : <ul>
            {jobs.map(j => <JobCard job={j} key={j.id} />)}
          </ul>
      }
    </section>
  );
}
