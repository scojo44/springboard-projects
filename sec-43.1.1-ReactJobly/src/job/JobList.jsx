import React from 'react'
import JoblyApi from '../api'
import useJoblyAPI from '../hooks/useJoblyAPI'
import LoadingIndicator from '../widgets/LoadingIndicator'
import Alert from '../widgets/Alert'
import SearchForm from '../widgets/SearchForm'
import JobCard from './JobCard'
import './JobList.css'

export default function JobList() {
  const {data: jobs, setParams, error, isLoading} = useJoblyAPI(JoblyApi.getJobs);

  if(isLoading) return <LoadingIndicator />;

  return (
    <section className="JobList">
      <SearchForm fieldName="titleLike" returnQuery={query => setParams(query)} />
      {error
        ? <Alert type="error" messages={[`Error loading jobs: ${error}`]} />
        : <ul>
            {jobs.map(j => <JobCard job={j} key={j.id} />)}
          </ul>
      }
    </section>
  );
}
