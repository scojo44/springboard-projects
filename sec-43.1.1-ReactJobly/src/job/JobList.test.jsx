import {render, waitForElementToBeRemoved} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import JoblyApi from '../api'
import JobList from './JobList'

const TEST_JOBS = [{
  id: 1,
  title: 'Test Engineer',
  salary: 200000,
  equity: .05,
  companyHandle: 'test-corp',
},
{
  id: 2,
  title: 'Test Developer',
  salary: 150000,
  equity: .1,
  companyHandle: 'test-llc',
}];

// Mock the API hook
JoblyApi.getJobs = vi.fn(() => TEST_JOBS);

describe('JobList Tests', () => {
  it('Renders without crashing', async () => {
    const {getByText} = render(
      <MemoryRouter>
        <JobList />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
  });

  it('Matches snapshot', async () => {
    const {asFragment, getByText} = render(
      <MemoryRouter>
        <JobList />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
    expect(asFragment()).toMatchSnapshot();
  });

  it('Lists some jobs', async () => {
    const {getByText} = render(
      <MemoryRouter>
        <JobList />
      </MemoryRouter>
    );

    await waitForElementToBeRemoved(getByText('Loading...'));
    expect(getByText(TEST_JOBS[0].title)).toBeInTheDocument();
    expect(getByText('200000')).toBeInTheDocument();
    expect(getByText('0.05')).toBeInTheDocument();
    expect(getByText(TEST_JOBS[1].title)).toBeInTheDocument();
    expect(getByText('150000')).toBeInTheDocument();
    expect(getByText('0.1')).toBeInTheDocument();
  });
});
