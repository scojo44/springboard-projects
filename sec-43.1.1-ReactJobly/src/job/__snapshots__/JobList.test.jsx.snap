// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JobList Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="JobList"
  >
    <form
      class="SearchForm"
    >
      <input
        id="titleLike"
        name="titleLike"
        placeholder="Enter search term"
        type="text"
        value=""
      />
      <button
        type="button"
      >
        Search
      </button>
    </form>
    <ul>
      <li
        class="JobCard"
      >
        <h3>
          Test Engineer
        </h3>
        <p>
          Salary: 
          <span
            class="detail-value"
          >
            200000
          </span>
        </p>
        <p>
          Equity: 
          <span
            class="detail-value"
          >
            0.05
          </span>
        </p>
      </li>
      <li
        class="JobCard"
      >
        <h3>
          Test Developer
        </h3>
        <p>
          Salary: 
          <span
            class="detail-value"
          >
            150000
          </span>
        </p>
        <p>
          Equity: 
          <span
            class="detail-value"
          >
            0.1
          </span>
        </p>
      </li>
    </ul>
  </section>
</DocumentFragment>
`;
